### DISCUSSION TITLE
Reverted back to previous version

### DISCUSSION THREAD

**Discussion by infromthecold** [maziac/DeZog#68] [General]
Thanks for the update, however I have reverted back to the previous version 2.3.1, as the changes you have made to the green button break my development flow. 

The old version I used to NMI+SYMBOL SHIFT then press the green arrow if the debugger was not running to start it, which would run my code without stopping at the entry point. Then if the debugger was running I would either stop it at a breakpoint using the game pad, or NMI+SYMBOL SHIFT and then click the green circle on the debugger panel and it would start a new session even if I had compiled a new version, which was nice and easy flow.  Sometimes I would press the green arrow by mistake which would crash the debugger forcing me to close the serial window and reopen it to regain debugging. 

However  with the latest version, everything is similar to above however sometimes the green circle on the debugger would restart the session, with the code stopping at the entry point meaning, the file window will change to the code entry point and I have to press continue . Also if I had made changes to my code, it crashes the serial again with a failure to connect. 

So can I have a switch so I can continue using the older system  please? 

Oh and this is what I am writing with your tools. 
https://luckyredfish.com/previews-of-tx-1696/

**Comment by maziac** [comment]
Looks great. It's always nice to get such feedback: that someone has use for the tool and creates something great with its help.

On your comments:
"Also if I had made changes to my code, it crashes the serial again with a failure to connect."
Crashing is a hard word :-) I added code to detect the situation that the file was newly compiled and throws an error.
For restart (green cycle) the labels are not re-read, that could lead in a situation that potentially (on a new compile) the labels are wrong.
I wanted to avoid that.
I have to think about this. Maybe I can re-read that labels, but could be that this conflicts with some other entity.
However, is it too much of work to press the red (stop) button and pressing continue (green triangle) instead of the green circle?
This should behave as before.

"code stopping at the entry point": Doesn't this depend on how you set it up? I.e. "startAutomatically" = true.
If you have set it to false please let me know. Then it's most probably a bug.

**Comment by infromthecold** [comment]
No obviously I could press stop and start again, but I thought that was the
point of the restart session button (green circle?), yes first run it will
not stop at the entry point, but the restart session does stop the at the
entry point.

😁

**Comment by maziac** [comment]
Please try the release here:
https://github.com/maziac/DeZog/releases/tag/v2.3.2-x
It should at least fix the startAutomatically issue.

**Comment by maziac** [comment]
From vscode point of view the difference between pressing stop/continue or restart is:
On restart the configuration (i.e. the launch.json) is not re-read.
Basically it is more lightweight, i.e. the socket connection does not need to be terminated/reconnected if the launch.json does not change.
Also the whole setup has been done already and can be skipped.
But problem remains if the sources are recompiled.

**Comment by maziac** [comment]
I compiled a new version (v2.3.2-2), please try:
https://github.com/maziac/DeZog/releases/edit/v2.3.2-x

It should now re-read the sld/list files, i.e. you should be able to re-compile your sources.
It also re-reads the launch-json which, personally, I think is a good thing (I myself was often surprised that a change in launch.json was not reflected on a restart, although I should have known better).

(On the other hand I dropped the idea that memory views and variable/expressions would remain on a restart.)
But, on the good side again, the socket connection is re-used.

**Comment by maziac** [comment]
Or better v2.3.2-3:
https://github.com/maziac/DeZog/releases/edit/v2.3.2-x

**Comment by infromthecold** [comment]
Sorry not had chance to test it yet.

**Comment by maziac** [comment]
Here is a pre-release for the 2.4.0:
https://github.com/maziac/DeZog/discussions/70
Unit testing is now integrated into DeZog.
